src/scripts/fetch-routes.ts

```diff
@@ -88,15 +100,17 @@ const fetchPaginatedSlugsForOperation = ({
 // fetches the total number of items for a given operation
 const fetchMetaForOperation = ({
   operation,
+  filter,
   locale,
 }: {
   operation: string;
+  filter: QueryFilter;
   locale: string;
 }) => {
   return datocmsFetch({
     query: `
         query Meta ($locale: SiteLocale) {
-            _${operation}Meta(locale: $locale) {
+            _${operation}Meta(locale: $locale, filter: ${toGqlFilter(filter)}) {
```

Comment:
I tried using graphql variables, but that requires us to specify a type for each route, which we can not derive from the 'operation' (because `Person` model becomes `allPeople` in plural form 🫠)